src/components/TokenBalance.tsx

```diff
@@ -87,20 +97,26 @@ function TokenBalance({
         <Image source={{ uri: tokensWithUsdValue[0].imageUrl }} style={styles.tokenImg} />
         <View style={styles.column}>
           <Text style={style} testID={'TotalTokenBalance'}>
-            {localCurrencySymbol}
-            {totalTokenBalanceLocal?.toFormat(2) ?? '-'}
-          </Text>
-          <Text style={styles.tokenBalance}>
-            {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
+            {(overrideHideBalance || !hideBalance) && localCurrencySymbol}
+            {hideBalance && !overrideHideBalance
+              ? 'XX' + decimalSeparator + 'XX'
+              : totalTokenBalanceLocal?.toFormat(2) ?? '-'}
           </Text>
+          {!hideBalance && (
```

Comment:
seems like overrideHideBalance is not honored here. Another reason to drop it and just have a single boolean prop for this function.